Trait Upports & Tweaks

Cameron-The-Raven opened the pull request:
## About The Pull Request
Upports a LOT of traits.


## Additions:

### Positives:
Master Linguist (Allows knowing EVEN MORE languages, up to 12 extra languages)

Dense Bones (Your bones are 50% more durable)

Pressure Perks: (These perks either decrease the amount of pressure needed to be safe by 33% or 100% (Low Pressure Min/Maj), or increase the safe pressure by 2x or 4x. (High Pressure Min/Maj). General Version increases high pressure by 3x and low pressure by 66%. General Maj increases high by 4x and min by 100%).)
(Additionally, low pressure lowers your breath pressure by 33% and general pressure lowers your breath pressure by 66%)

Low Pressure Minor (Decreases safe low pressure requirements by 33%. Decreases breath pressure by 33%)
Low Pressure Major  (Decreases safe low pressure requirements by 100%. Decreases breath pressure by 66%)
High Pressure Minor (Increases safe max pressure by 2x)
High Pressure Major (Increases safe max pressure by 4x)
General Pressure Resistance (Decreases safe low pressure requirements by 66% and increases high pressure by 3x Decreases breath pressure by 66%))
General Pressure Resistance Major (Decreases safe low pressure requirements by 100% and increases high pressure by 4x. Decreases breath pressure by 66%)

Blood Volume, High (Blood volume increased by 50%)
Blood Volume, Very High (Blood volume increased by 150%)

Heavyweight (Stuns and weakens are 50% lessened on you. You also count as LARGE for mob interactions and HEAVY when bumped into)

Absorbent (You clean clothing you are wearing, items you touch, the floor you walk on, and items you walk over.)

Adrenaline Rush (When in critical condition (HP < 0%) you gain an Adrenaline Rush, making you immune to disables, immune to pain, increase your health slightly, take less damage, lowered o2 damage, hit harder, hit faster, move MUCH faster, at the cost of bleeding out faster and metabolizing faster. This can only happen once every 30 minutes.)

Burn Resist, Major (Burn resist, but 40% resistance)
Brute Resist, Major (Brute resist, but 40% resistance)

High Endurance, Major (Your max HP is now 150, giving you 300 effective HP)
High Endurance, Extreme (Your max HP is now 175, giving you 350 effective HP)

Pain Tolerance, Minor (Pain is 10% less effective on you)
Pain Tolerance (Pain is 20% less effective on you)
Pain Tolerance, Major (Pain is 40% less effective on you)

Improved Biocompatibility (Medication works 20% MORE effective on you)
Photoresistance, Major (Flashes are 50% less effective on you)

Hardy, Extreme (You have no slowdown with equipment)

Evolved Bloodsucker (You are a bloodsucker, but food is 50% effective for you as well)

Robust Gut (You can eat food that has toxins in it and be fine. Injected toxins will still harm you.)

Breathless (You don't breathe in the 'natural' sense of things and don't have lungs. See: Plants)

Light Breather (Your minimal breath pressure is 12 KPA)

EMP Resistance (If you are a protean, you take 30% less damage and effects from EMPs. For everyone else, EMPs on your objects/limbs/organs count as one degree lessened)
Major EMP Resistance (Same as above, but 50% and 2 degrees lessened)

Grappling Expert (Your grabs are harder to escape from and escaping others grabs is easier.)

### Negatives: 
Carbon Dioxide Breather (plants!)
Deep Sleeper (Wake up 4x slower from sleep)
Deep Breather (Require 2kpa more air to breathe) 


Faulty Wires (Your frame now has an EMP sensitivity and you take 130% strength from effects (Meaning EMPs: Blind, deafen, damage, and confuse you). All items on you when EMP'd are affected as if the EMP is 1 strength higher than initial.)

Poor Construction (The same as above, but 160%, EMPs are 2 stronger, and EMPs weaken you.)

Obligate Pain Simulation (Like synthetic cosmetic pain, but can't be turned off!)

Water Weakness (Water damages you! Only take 0.3 damage per affliction if fully naked. Takes clothing reagent permeability into account)

Extreme Water Weakness (Same as above, but max of 0.8 damage per affliction)

Lesser Lightweight (A slightly less intense version of lightweight. Only has you get knocked over if you're unprepared (green intent))

Scrawny (Grabs are harder to break out of and others can break from your grabs easier.)

Insect Sting (Gives you a proc that lets you inject a nearby entity with 'Irritant toxin' which is painful, does minor toxins, and makes your eyes blurry)

DISABLED Negative: Episodic Hallucinations (Makes you occasionally have hallucinations. Requires you take tercozolam, which was added in this PR as well.)

### Neutrals:
Hard Feet is now a neutral. (This trait is practically cosmetic, as you can just 'get this trait' by putting a pair of shoes on.)

Hide is now a neutral. (This is more of an RP trait. It allows you to hide under objects, but is completely nullified by AR huds and the such.) Hopefully, this leads to more 'surprise vore' shenanigans.

Adds Singularity Metabolism. (You get hungry REALLY fast)

Adds Bitesize Traits (You eat more/less per bite!)

Adds Slippery Reflexes (When slipping, you land on your feet. This has a short cooldown, so if you slip again you'll still slide and eat dirt.)

## Tweaks:
Metabolism perks can be taken by synths
Darksight text had some slight grammar changes. Also tells you it's 'significantly further' instead of 'short distance'
Darksight Major text had some slight grammar changes. Also tells you it's the full screen instead of 'almost the whole screen'
Adds Sonar Ping proc (stand still and hear if something is coming) to Vibration Sense

## Misc:
Makes Absorbent into a component
Makes low blood sugar into a component
Makes Burninlight use (un)RegisterWithParent
Makes Crowd Detcetion use (un)RegisterWithParent
Makes Drippy use (un)RegisterWithParent
Makes Photosynthesis use (un)RegisterWithParent
Makes Weaver us (un)RegisterWithParent
Fixes Gargoyle unregistering a signal it doesn't have registered.
Fixes brute resist to block brute resist perks
Fixes burn resist to block burn resist perks
Makes Haemophilia , Major use bloodloss_rate like Haemophilia instead of a modifier.

TODO:

- [x] Check Absorbent
- [x] Check Low Blood Sugar
- [x] Check Burninlight
- [x] Check Crowd Detection
- [x] Check Drippy
- [x] Check Photosynthesis
- [x] Check Weaver


## Changelog
:cl: Diana
qol: Darkvision positive no longer gives you flash weakness negative
qol: Metabolism perks can be taken  by synths AND humans. Synths can now metabolize beer faster! 
qol: Hard feet is now a neutral instead of a positive trait. It cost 0 points and is so unimpactful that it made little sense to have it take a positive slot when it's mostly just a fluff trait.
fix: Brute resistance positives block you from taking other brute resistance positives instead of fire resistance. The opposite is true.
spellcheck: Darkvision note adjustments.
add: A LOT of new traits have been added from downstream.
add: Hide added as a neutral trait. AR huds make this irrelevant so eh. Could lead to stealthy pred stuff.
/:cl:

SatinIsle commented:
I think all of this is pretty cool, except the changes to dark vision flash mod and range. I think the minor debuff to flashing resistance is good flavour and should be kept.

Cameron-The-Raven commented:
> I think all of this is pretty cool, except the changes to dark vision flash mod and range. I think the minor debuff to flashing resistance is good flavour and should be kept.

I'll go ahead and edit the dark vision to prior version when I work some more on this PR

Cameron-The-Raven commented:
> I think all of this is pretty cool, except the changes to dark vision flash mod and range. I think the minor debuff to flashing resistance is good flavour and should be kept.

Reverted the dark vision changes back to prior. Updated the description to be more accurate (8 tiles informs you it's the full screen)

Cameron-The-Raven commented:
Absorbent works 
![dreamseeker_2025-10-24_16-07-44](https://github.com/user-attachments/assets/f46b34b7-5f7e-4888-84c2-b0ff4055d37a)

All the rest of them also worked when tested (photosynthesis is laughably bad but it works, making it stronger is for another PR)

So...This PR is good for code review now.

Kashargul commented on code/datums/components/traits/low_sugar.dm:
```diff
@@ -0,0 +1,38 @@
+		return
+	if((living_guy.nutrition < nutrition_threshold) && prob(5))
+		if(living_guy.nutrition > nutrition_weak)
+			to_chat(living_guy,span_warning("You start to feel noticeably weak as your stomach rumbles, begging for more food. Maybe you should eat something to keep your blood sugar up"))
```
- [ ] diana can we please use spaces after , xS it's everywhere shoved together
